src/uu/dd/src/dd.rs

```diff
@@ -960,9 +965,6 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         #[cfg(unix)]
         Some(ref outfile) if is_fifo(outfile) => Output::new_fifo(Path::new(&outfile), &settings)?,
         Some(ref outfile) => Output::new_file(Path::new(&outfile), &settings)?,
-        None if is_stdout_redirected_to_seekable_file() => {
-            Output::new_file(Path::new(&stdout_canonicalized()), &settings)?
-        }
```

Comment:
I believe this workaround is now handled more appropriately in the `new_stdout()` case.

Reply:
@jfinkels ping ? :)

Reply:
> But the seeking is different now, isn't it? Because `StdoutFile` copies `0`'s instead of seeking.

Hmm, yes I guess this removal was premature. It seems like there should be a way to unify these two code paths in a cleaner way, but maybe I'll just postpone that until a later pull request. For example, maybe the decision about whether the destination is seekable could move inside the `new_stdout()` constructor.

Reply:
Oh actually, I forgot---if I leave this branch in, then the new test case fails:
```
$ printf "abcde\n" | (dd bs=1 skip=1 count=1 && dd bs=1 skip=1)  2> /dev/null
bde
```
(this comes from the GNU test suite). For reference, we added the `if is_stdout_redirected_to_seekable_file()` in pull request #3880.

Let me see if there's a way to reconcile the two.

Reply:
I just updated this branch to keep the `None if is_stdout_redirected_to_seekable_file() => {` branch. There seems to be an issue with the new test case I added, but I'm not sure what the problem is.

When I run the test case from the command line I get the expected behavior:
```
$ printf "abcde\n" | (./target/debug/dd bs=1 skip=1 count=1; ./target/debug/dd bs=1 skip=1) 2> /dev/null
bde
```
but when I run `cargo test` I get a failure:
```
$ cargo test test_dd::test_multiple_processes_writing_stdout
[...]
---- test_dd::test_multiple_processes_writing_stdout stdout ----
run: /bin/sh -c /home/jeffrey/src/coreutils/target/debug/coreutils printf 'abcde
' | ( /home/jeffrey/src/coreutils/target/debug/coreutils dd bs=1 skip=1 count=1; /home/jeffrey/src/coreutils/target/debug/coreutils dd bs=1 skip=1 ) 2> /dev/null
thread 'test_dd::test_multiple_processes_writing_stdout' panicked at 'assertion failed: `(left == right)`

Diff < left / right > :
<de
>bde
```